runtime/autoload/provider/script_host.py

```diff
@@ -198,22 +204,25 @@ def _find_module(fullname, oldtail, path):
             name = oldtail[:idx]
             tail = oldtail[idx+1:]
             fmr = imp.find_module(name, path)
-            module = imp.load_module(fullname[:-len(oldtail)] + name, *fmr)
+            module = imp.find_module(fullname[:-len(oldtail)] + name, *fmr)
             return _find_module(fullname, tail, module.__path__)
         else:
-            fmr = imp.find_module(fullname, path)
-            return imp.load_module(fullname, *fmr)
+            return imp.find_module(fullname, path)
 
     class VimModuleLoader(object):
         def __init__(self, module):
             self.module = module
 
         def load_module(self, fullname, path=None):
-            return self.module
+            # Check sys.modules, required for reload (see PEP302).
+            if fullname in sys.modules:
+                return sys.modules[fullname]
+            return imp.load_module(fullname, *self.module)
 
     class VimPathFinder(object):
         @classmethod
```

Comment:
This should probably become an [`@staticmethod`](https://docs.python.org/3/library/functions.html#staticmethod), since the `cls` parameter is not used at all. Same with `find_spec`.